javascript/practice/app.js

```diff
@@ -0,0 +1,140 @@
+// Q1 変数
+// 変数 nickname, age に自分の名前と年齢を代入し、文字列連結でコンソールに出力。
+let nickname = 'なっちゃん';
+let age = '26歳';
+let greet = '私のニックネームは' + nickname + 'です。' + '年齢は' + age + 'です。';
+console.log(greet);
+
+// Q2 配列
+// 変数 languages に JavaScript, PHP, Ruby, Python, Go の 5 つの値を持つ配列を代入し、その中から JavaScript, Python を取得し、テンプレートリテラルを用いてコンソールに出力。
+let languages = ['JavaScript','PHP','Ruby','Python'];
+console.log('languages[0] =>', languages[0]);
+console.log('languages[3] =>', languages[3]);
+
+let favorite = `私の好きな言語は${languages[0]}です。次は${languages[3]}を勉強してみたいです。`
+console.log(favorite);
+
+
+// Q3 オブジェクト
+// 下記の変数 user を使用し 26 をコンソールに出力。
+let user = {
+  name: 'John',
+  age: 26,
+  bloodType: 'A',
+  favorite: 'card',
+};
+
+console.log(user.age);
+
+
+// Q4 配列 × オブジェクト
+// 下記の変数 playerList を使用し The Legend of Zelda をコンソールに出力。
+let playerList = [
+  {
+    name: 'John',
+    age: 26,
+    favorites: ['Card Game', 'Basket Ball', 'Programming'],
+  },
+  {
+    name: 'Bob',
+    age: 33,
+    favorites: ['Tinder', 'The Legend of Zelda'],
+  },
+  {
+    name: 'Michael',
+    age: 22,
+    favorites: ['Football', 'Smash Bros.'],
+  },
+];
+
+console.log(playerList[1].favorites[1]);
+
+
+// Q5 四則演算
+// Q4 で定義した変数 playerList を使用し、John, Bob, Michael の年齢の平均値を計算した結果をコンソールに出力。
+const arr = [26, 33, 22];
+let sum = 0;
+for (let i=0; i<arr.length; i++) {
+  sum += arr[i];
+}
+console.log(sum / arr.length);
+
+
+// Q6 関数
+// Hello とコンソールに出力する関数 sayHello を定義し、実行。
+// 変数 sayWorld に World とコンソールに出力する無名関数を代入し、実行。
+function sayHello() {
+  console.log("Hello");
+}
+
+sayHello();
+
+const myFunction = function(sayHello) {
+  return `World、 ${sayHello}`
+};
+
+const greetingMessage = myFunction('World');
+console.log(greetingMessage); 
+
+
+// Q7 メソッド
+// Q3 で定義した変数 user に '2000-09-27'を値とする birthday プロパティを追加。
+// Q3 で定義した変数 user に Hello！とコンソールに出力するメソッド sayHello を追加し、実行。
+user.birthday = '2000-09-27';
+console.log(user);
+
+user.sayHello = function(){
+  console.log('Hello!');
+}
+
+sayHello();
+
+// Q8 引数
+// 2 つの引数 x, y の和をコンソールに出力する add メソッドを定義し、7 になるように実行。
+// 2 つの引数 x, y の差をコンソールに出力する subtract メソッドを定義し、10 になるように実行。
+// 2 つの引数 x, y の積をコンソールに出力する multiply メソッドを定義し、49 になるように実行。
+// 2 つの引数 x, y の商をコンソールに出力する divide メソッドを定義し、5 になるように実行。
+
+let calc = {};
+
+function add(obj, x, y) {
+  obj.add = x + y;
+}
+add(calc, 6, 1);
+
+function subtract(obj, x, y) {
+  obj.subtract = x - y;
+}
+subtract(calc, 15, 5);
+
+function multiply(obj, x, y) {
+  obj.multiply = x * y;
+}
+multiply(calc, 7, 7);
+
+function divide(obj, x, y) {
+  obj.divide = x / y;
+}
+divide(calc, 25, 5);
+
+console.log(calc)
+
+// Q9 返り値
+// 2 つの引数 x, y を受け取り、x を y で割った剰余を返り値とする関数 remainder を定義し、5 と 3 を引数に渡して実行した返り値を用いて文字列連結で以下のようにコンソールに出力。
+function reminder(x,y) {
+  let divide = x % y;
+  return divide;
+}
```

Comment:
[FYI]
練習問題Q9

こちらは1行でもこんな形で書くことができます！

```suggestion
function reminder(x,y) {
  return x % y;
}
```